src/coreclr/vm/interpexec.cpp

```diff
@@ -2053,7 +2068,7 @@ void InterpExecMethod(InterpreterFrame *pInterpreterFrame, InterpMethodContextFr
                             pInterpreterFrame->SetTopInterpMethodContextFrame(pFrame);
                             GCX_PREEMP();
                             // Attempt to setup the interpreter code for the target method.
-                            if ((targetMethod->IsIL() || targetMethod->IsNoMetadata()) && !targetMethod->IsUnboxingStub())
+                            if (targetMethod->CanBeInterpreted())
```

Comment:
Nit: I would not worry about factoring this out into a method. This whole logic is questionable and very slow in the presence of interpreter -> native transitions that will be very common in the shipping configurations. This code will need to be redone, and I do not expect that we will need CanBeInterpreted condition once that happens.